homework5/oop_1.py

```diff
@@ -0,0 +1,131 @@
+"""
+Необходимо создать 3 класса и взаимосвязь между ними (Student, Teacher,
+Homework)
+Наследование в этой задаче использовать не нужно.
+Для работы с временем использовать модуль datetime
+
+1. Homework принимает на вход 2 атрибута: текст задания и количество дней
+на это задание
+Атрибуты:
+    text - текст задания
+    deadline - хранит объект datetime.timedelta с количеством
+    дней на выполнение
+    created - c точной датой и временем создания
+Методы:
+    is_active - проверяет не истекло ли время на выполнение задания,
+    возвращает boolean
+
+2. Student
+Атрибуты:
+    last_name
+    first_name
+Методы:
+    do_homework - принимает объект Homework и возвращает его же,
+    если задание уже просрочено, то печатет 'You are late' и возвращает None
+
+3. Teacher
+Атрибуты:
+     last_name
+     first_name
+Методы:
+    create_homework - текст задания и количество дней на это задание,
+    возвращает экземпляр Homework
+    Обратите внимание, что для работы этого метода не требуется сам объект.
+
+PEP8 соблюдать строго.
+Всем перечисленным выше атрибутам и методам классов сохранить названия.
+К названием остальных переменных, классов и тд. подходить ответственно -
+давать логичные подходящие имена.
+"""
+import datetime
+
+
+class Homework:
+    """
+    Homework takes two attributes: task's text and number of days for performing task
+
+    Attributes:
+        text - task's text
+        deadline - stores a datetime.timedelta object with the number of
+    days to execute
+        created - with date and time of creation
+
+    Method:
+        is_active - check deadline and return boolean
+    """
+
+    def __init__(self, text, deadline, created=datetime.datetime.now()):
+        self.text = text
+        self.deadline = datetime.timedelta(deadline)
+        self.created = datetime.datetime.now()
+
+    def is_active(self):
+        time_now = datetime.datetime.now()
+        time_deadline = self.created + self.deadline
+        return False if time_now >= time_deadline else True
+
+
+class Student:
+    """
+    Student takes two attributes: last_name and first_name
+
+    Attributes:
+        last_name
+        first_name
+
+    Method:
+        do_homework - takes class Homework and check is active homework or not
+    """
+
+    def __init__(self, last_name, first_name):
+        self.last_name = last_name
+        self.first_name = first_name
+
+    @classmethod
+    def do_homework(cls, instance):
+        method_is_active = Homework.is_active(instance)
```

Comment:
Надо проверять активна ли конкретная работа, которая передана в функцию, т.е.:

```suggestion
        method_is_active = instance.is_active(instance)
```